app/controllers/profiles_controller.rb

```diff
@@ -1,10 +1,25 @@
+require 'lol'
+require 'certified'
+require 'ostruct'
 class ProfilesController < ApplicationController
+  attr_accessor :summonerObj, :summonerReq, :statsObj, :statsReq, :champsReq, :champsObj
+  before_action :verify_summoner_name, only: [:create]
+  before_action :get_summoner, only: [:show]
+  before_action :get_summoner_ranked_stats, only: [:show]
+  before_action :get_champion_request, only: [:show]
+  before_action :get_overall_WLR, only: [:show]
+  before_action :get_overall_KDR, only: [:show]
+  before_action :get_highest_WLR_champ, only: [:show]
+  before_action :get_highest_KDR_champ, only: [:show]
+  before_action :get_most_played_champ, only: [:show]
+
```

Comment:
instead of having all these `before_action` filters, you could just have one and then call the remaining functions from inside that one function.  This is actually more efficient as well since then rails doesn't have to analyze all these filters everytime it does an action.